src/core/message.ts

```diff
@@ -115,13 +185,27 @@ export class DefaultMessage implements Message {
     public readonly createdAt: Date,
     public readonly metadata: MessageMetadata,
     public readonly headers: MessageHeaders,
+    public readonly deletedAt?: Date,
```

Comment:
One for you and @vladvelici to discuss where it belongs - we need to do the CHADR-064 update to timeserials here.

Reply:
I think we can make a PR for that separately and merge it as soon as we get it done?